File: src/main/java/io/github/archipelagominecraft/core/loaders/legacy/LegacyForgeCorePlugin.java
//? if <=1.12.2 {
/*package io.github.archipelagominecraft.core.loaders.legacy;

import com.google.common.collect.ImmutableList;
import io.github.archipelagominecraft.core.ArchipelagoClientConstants;
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;
import zone.rong.mixinbooter.IEarlyMixinLoader;

import javax.annotation.Nullable;
import java.util.Collections;
import java.util.List;
import java.util.Map;

// For MixinBooter

public class LegacyForgeCorePlugin implements IFMLLoadingPlugin, IEarlyMixinLoader {

    @Override
    public String[] getASMTransformerClass() {
        return new String[0];
    }

    @Override
    public String getModContainerClass() {
        return null;
    }

    @Nullable
    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) {

    }

    @Override
    public String getAccessTransformerClass() {
        return null;
    }

    @Override
    public List<String> getMixinConfigs() {
        return ImmutableList.of(ArchipelagoClientConstants.MOD_MIXINS_FILE + ".mixins.json");
    }
}

*///?}

File: mod/src/main/kotlin/io/github/archipelagominecraft/core/loaders/legacyforge/LegacyForgeCorePlugin.kt
//? if <=1.12.2 {
/*package io.github.archipelagominecraft.core.loaders.legacyforge

import io.github.archipelagominecraft.core.ArchipelagoClientConstants


//? if >= 1.8.9 {
import zone.rong.mixinbooter.IEarlyMixinLoader
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin
//? } else {

/*import cpw.mods.fml.relauncher.IFMLLoadingPlugin
import com.gtnewhorizon.gtnhmixins.IEarlyMixinLoader
import io.github.archipelagominecraft.core.mixins.ForceSeedMixinLegacy

*///? }

class LegacyForgeCorePlugin : IFMLLoadingPlugin, IEarlyMixinLoader {
    override fun getASMTransformerClass(): Array<String> = emptyArray()

    override fun getModContainerClass(): String? = null

    override fun getSetupClass(): String? = null

    override fun injectData(data: MutableMap<String, Any>) {}

    override fun getAccessTransformerClass(): String? = null

    val mixinConfigFile = ArchipelagoClientConstants.MOD_MIXINS_FILE_PREFIX + ".mixins.json"

    init {
        println("LEGACY FORGE CORE PLUGIN LOADED")
    }
//? if >= 1.8.9 {
    override fun getMixinConfigs(): List<String?> =
        listOf(mixinConfigFile)
    //?} else {
    /*override fun getMixinConfig(): String = mixinConfigFile
    override fun getMixins(loadedCoreMods: Set<String>): List<String> = listOf(
        ForceSeedMixinLegacy::class.simpleName!!
    )
    *///? }
}
*///?}

